### PR TITLE
Added caching method for application redirect context

### PR DESCRIPTION
**Pull request by lalimasharda**
- Cache the Application Id of the application requesting redirect in order to apply the correct context for a successful login.

### CODE REVIEW

File: lib/msal-browser/src/cache/BrowserCacheManager.ts [hunk 1]
```diff
@@ -1062,6 +1062,16 @@ export class BrowserCacheManager extends CacheManager {
...
     }
+
+    getRedirectRequestContext(): string | null {
+        const key = `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REDIRECT_CONTEXT}`;
+        return this.getTemporaryCache(key);
+    }
+     
+    setRedirectRequestContext(value: string): void {
+        const key = `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REDIRECT_CONTEXT}`;
+        this.setTemporaryCache(key, value, false);
```
Comment (tnorling) [inline lib/msal-browser/src/cache/BrowserCacheManager.ts]: I think we do want to generate the key
```suggestion
this.setTemporaryCache(TemporaryCacheKeys.REDIRECT_CONTEXT, value, true);
```

File: lib/msal-browser/src/cache/BrowserCacheManager.ts [hunk 2]
```diff
@@ -1062,6 +1062,16 @@ export class BrowserCacheManager extends CacheManager {
 
         return currentCacheKey;
     }
+
+    getRedirectRequestContext(): string | null {
+        const key = `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REDIRECT_CONTEXT}`;
+        return this.getTemporaryCache(key);
```
Comment (tnorling) [inline lib/msal-browser/src/cache/BrowserCacheManager.ts]:
```suggestion
        return this.getTemporaryCache(TemporaryCacheKeys.REDIRECT_CONTEXT, true);
```

File: change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Merge branch 'dev' of https://github.com/AzureAD/microsoft-authentication-library-for-js into lalimas/browsercachingupdates",
```
Comment (tnorling) [inline change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json]: Can you give this a more descriptive comment?

**Review by tnorling — COMMENTED** [review]
Can you also add some quick tests for the new methods?

File: change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json (hunk 3 above)
Comment (lalimasharda) [reply change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json]: Done

File: lib/msal-browser/src/cache/BrowserCacheManager.ts (hunk 2 above)
Comment (lalimasharda) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts]: Done

File: lib/msal-browser/src/cache/BrowserCacheManager.ts (hunk 1 above)
Comment (lalimasharda) [reply lib/msal-browser/src/cache/BrowserCacheManager.ts]: Done

File: change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Added caching methods to set or get redirect context of application during redirect flow.",
```
Comment (tnorling) [inline change/@azure-msal-browser-54359286-3f28-4e94-bee5-d6a43829a361.json]:
```suggestion
  "comment": "Added caching methods to set or get redirect context of application during redirect flow. #5411",
```

**Review by tnorling — APPROVED** [review]
LGTM

**Review by sameerag — APPROVED** [review]

**Pull request merged**